### PR TITLE
Qdm 5.4

### PR DESCRIPTION
**Pull request by mayerm94**
Use new model-info file to generate models, add ratio and fix lingering ID issues that surfaced through the addition of prescriberId and dispenserId attributes to medication datatypes

- [x] UPDATE DIST AFTER https://github.com/cqframework/cql-execution/pull/36 IS PUBLISHED. Can point to local cql-execution branch, but updating dist while pointing to a git branch in package.json fails

Pull requests into cqm-models require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/BONNIE-1605
- [x] Internal ticket links to this PR 
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] If there were any JavaScript changes, this PR has updated the `dist` directory using `npm run dist`
- [x] If applicable, the library version number in `package.json` and `cqm-models.gemspec` has been updated
- [x] All changes can be reproduced by running the generator script

**Bonnie Reviewer:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code
- [x] You have executed the generator script, and made sure no changes were made that the generator did not reproduce itself

**Cypress Reviewer:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code
- [x] You have executed the generator script, and made sure no changes were made that the generator did not reproduce itself

### CODE REVIEW

File: app/models/qdm/basetypes/ratio.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    # Converts an object of this instance into a database friendly value.
+    def mongoize
+      { numerator: @numerator, denominator: @denominator, _type: 'QDM::Ratio' }
+    end
+
+    class << self
+      # Get the object as it was stored in the database, and instantiate
+      # this custom class from it.
+      #
+      # The array elements in demongoize are the same 5 elements used in mongoize, i.e.
```
Comment (losborne) [inline app/models/qdm/basetypes/ratio.rb]: comment says 5 but there are only 2 elements

File: app/models/qdm/basetypes/ratio.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      end
+
+      # Takes any possible object and converts it to how it would be
+      # stored in the database.
+      def mongoize(object)
+        case object
+        when nil then nil
+        when QDM::Ratio then object.mongoize
+        when Hash
+          object = object.symbolize_keys
```
Comment (losborne) [inline app/models/qdm/basetypes/ratio.rb]: Is it possible to just call `demongoize(object).mongoize` here?  These 2 lines seem somewhat repeated from lines 25-26

calling "demongoize" from within "mongoize" does seem a bit strange, so maybe it could be pulled out into a `from_hash` or something like that?

Comment (mayerm94) [reply app/models/qdm/basetypes/ratio.rb]: All of the other basetypes behave the same way. I'd rather have ratio match the other basetypes since they have already proven themselves as functional.

File: app/models/models.rb [hunk 3]
```diff
@@ -21,7 +21,6 @@ module QDM
 
 # Generated models
 require_relative 'qdm/patient'
-require_relative 'qdm/ratio'
```
Comment (dczulada) [inline app/models/models.rb]: required_relative for 'qdm/basetypes/ratio' needs to be added

**Comment by dczulada** [comment]
Still looks good.
